.github/workflows/all-post-commit-workflows.yaml

```diff
@@ -27,6 +33,10 @@ jobs:
   eager-package-main:
     uses: ./.github/workflows/eager-package-main.yaml
     secrets: inherit
+  build-docs:
+    needs: [fd-unit-tests, sd-unit-tests]
```

Comment:
do you need to wait for these steps to finish? Given that you still need to compile right now on GH runners, the post-commit time is now quite a bit longer (1h 24min in your run)

Reply:
I can revert not to block it on the successful tests, but then we would be publishing the docs even if the tests are failing. I'll make the change to not gate on the tests.

Reply:
think we can keep this check after it uses the build artifact